libsolidity/ast/Types.h

```diff
@@ -899,7 +899,9 @@ class FunctionType: public Type
 		AddMod, ///< ADDMOD
 		MulMod, ///< MULMOD
 		ArrayPush, ///< .push() to a dynamically sized array in storage
+		ArrayPop, ///< .pop() from a dynamically sized array in storage
 		ByteArrayPush, ///< .push() to a dynamically sized byte array in storage
+		ByteArrayPop, ///< .pop() from a dynamically sized byte array in storage
```

Comment:
Actually, since we check the type of the array anyway, we do not have to introduce two different pop types.